spring/src/main/java/net/notejam/spring/user/forgot/RecoveryToken.java

```diff
@@ -0,0 +1,102 @@
+package net.notejam.spring.user.forgot;
+
+import java.time.Instant;
+
+import javax.persistence.Entity;
+import javax.persistence.ManyToOne;
+import javax.validation.constraints.Future;
+import javax.validation.constraints.NotNull;
+
+import org.springframework.data.jpa.domain.AbstractPersistable;
+
+import net.notejam.spring.user.User;
+
+/**
+ * The token to authorize a password recovery request.
+ *
+ * @author markus@malkusch.de
+ * @see <a href="bitcoin:1335STSwu9hST4vcMRppEPgENMHD2r1REK">Donations</a>
+ */
+@Entity
+public class RecoveryToken extends AbstractPersistable<Integer> {
+
+    private static final long serialVersionUID = 5923083445165411558L;
+
+    /**
+     * The user.
+     */
+    @NotNull
+    @ManyToOne
+    private User user;
+
+    /**
+     * The token.
+     */
+    @NotNull
+    private String token;
+
+    /**
+     * The expiration date.
+     */
+    @NotNull
+    @Future
+    private Instant expiration;
+
+    /**
+     * Returns the date when the token expires.
+     *
+     * @return The expiration date.
+     */
+    public Instant getExpiration() {
+        return expiration;
+    }
+
+    /**
+     * Sets the date when the token expires.
+     *
+     * @param expiration
+     *            The expiration date.
```

Comment:
This the default formatting when I let my IDE format the project. The format is called "Java Conventions" in Eclipse. It doesn't violate those 15 years old guidelines from Sun so it's convenient for me.